dds/DCPS/MultiTopicDataReader_T.cpp

```diff
@@ -283,18 +293,26 @@ MultiTopicDataReader_T<Sample, TypedDataReader>::cross_join(
       SampleVec resulting;
       for (typename SampleVec::iterator i = iterPR->second.begin();
         i != iterPR->second.end(); ++i) {
-        join(resulting, *i, no_keys, NULL, qp.data_reader_, other_meta);
+        DDS::ReturnCode_t ret = join(resulting, *i, no_keys, NULL, qp.data_reader_, other_meta);
```

Comment:
```suggestion
        const DDS::ReturnCode_t ret = join(resulting, *i, no_keys, 0, qp.data_reader_, other_meta);
```

And the rest of the return codes.